luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs

```diff
@@ -71,36 +95,96 @@ impl Component for WysiwygEditor<'_> {
+                                    if context.moving_with == MovingWith::Mouse {
+                                        context.end_global_xy = event.global_xy;
+                                    }
                                 }
                             });
                         }
                     }
                     Event::MouseUp { event } => {
                         if let Some(Dragging::Mover { mut context }) = dragging.deref() {
-                            if let Some(graphic_index) = editing_image_index {
-                                context.end_global_xy = event.global_xy;
-
-                                let (_, graphic) = screen_graphics
-                                    .iter()
-                                    .find(|(index, _)| index == &graphic_index)
-                                    .unwrap();
-                                let circumscribed =
-                                    context.move_circumscribed(graphic.circumscribed());
-
-                                SEQUENCE_ATOM.mutate(move |sequence| {
-                                    sequence.update_cut(
-                                        cut_id,
-                                        CutUpdateAction::ChangeGraphicCircumscribed {
-                                            graphic_index,
-                                            circumscribed,
-                                        },
-                                    )
-                                });
+                            if context.moving_with == MovingWith::Mouse {
+                                if let Some(graphic_index) = editing_image_index {
+                                    context.end_global_xy = event.global_xy;
+                                    update_sequence_with_mover(context, graphic_index);
+                                }
                             }
                         }
                         set_dragging.set(None);
                     }
+                    Event::KeyDown { event } => {
+                        let arrow_key_down = matches!(
+                            event.code,
+                            Code::ArrowUp | Code::ArrowLeft | Code::ArrowRight | Code::ArrowDown
+                        );
+                        if let Some(graphic_index) = *editing_graphic_index {
+                            if cut_editor_focused && arrow_key_down {
+                                let set_new_dragging = || {
+                                    let moving_with = match event.code {
+                                        Code::ArrowLeft => MovingWith::KeyLeft,
+                                        Code::ArrowRight => MovingWith::KeyRight,
+                                        Code::ArrowUp => MovingWith::KeyUp,
+                                        Code::ArrowDown => MovingWith::KeyDown,
+                                        _ => unreachable!(),
+                                    };
```

Comment:
I guess this mapping can be done before if statement so it can be used at 141 line. let's remove ugly unreachable! at 146.

Reply:
Fixed with cd51a27